### PR TITLE
fix: chat_models Qianfan not compatiable with SystemMessage

### PR DESCRIPTION
**Pull request by danielhjz**
- **Description:** QianfanEndpoint bugs for SystemMessages. When the `SystemMessage` is input as the messages to `chat_models.QianfanEndpoint`. A `TypeError` will be raised.
  - **Issue:** #10643
  - **Dependencies:** 
  - **Tag maintainer:** @baskaryan
  - **Twitter handle:** no

### CODE REVIEW

**Comment by NuODaniel** [comment]
@baskaryan CC
fix for compatibility to SystemMessage.

File: libs/langchain/langchain/chat_models/baidu_qianfan_endpoint.py [hunk 1]
```diff
@@ -174,9 +175,35 @@ def _convert_prompt_msg_params(
...
+            resulting dictionary.
+
+        Returns:
+            Dict[str, Any]: A dictionary containing the message content and default
+            parameters.
+
+        """
+        messages_dict = {
+            "messages": [
+                convert_message_to_dict(m)
```
Comment (baskaryan) [inline libs/langchain/langchain/chat_models/baidu_qianfan_endpoint.py:195]: can we just update this method

Comment (NuODaniel) [reply libs/langchain/langchain/chat_models/baidu_qianfan_endpoint.py:195]: That might be a difference between Qianfan API and Open-ai API. In Qianfan API, there is a field called `system`, which has the same level with `messages`. For example:

```
{
    "messages":[
        {
            "role":"user",
            "content":"Who are you?"
        }
    ],
    "system":"You are an artificial intelligence assistant. You were developed by Baidu Company, and you can provide assistance, advice, and chat ability to users. "
}
```

Comment (NuODaniel) [reply libs/langchain/langchain/chat_models/baidu_qianfan_endpoint.py:195]: > That might be a difference between Qianfan API and Open-ai API. In Qianfan API, there is a field called `system`, which has the same level with `messages`. For example:
> 
> ```
> {
>     "messages":[
>         {
>             "role":"user",
>             "content":"Who are you?"
>         }
>     ],
>     "system":"You are an artificial intelligence assistant. You were developed by Baidu Company, and you can provide assistance, advice, and chat ability to users. "
> }
> ```
> 
> So only updating the method `convert_message_to_dict` might be difficult. @baskaryan FYI

CC, if that OK for merge? @baskaryan

**Comment by NuODaniel** [comment]
ANY ONE PTAL @baskaryan, @eyurtsev, @hwchase17. thanks.

File: libs/langchain/langchain/chat_models/baidu_qianfan_endpoint.py (hunk 1 above)
Comment (baskaryan) [reply libs/langchain/langchain/chat_models/baidu_qianfan_endpoint.py:195]: makes sense, thank you @NuODaniel!

**Pull request merged**